File: sportsbook-app/odds-service/src/main/java/com/sportsbook/oddsservice/websocket/OddsUpdateBroadcaster.java
package com.sportsbook.oddsservice.websocket;

import com.sportsbook.events.OddsUpdated;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class OddsUpdateBroadcaster {

    private final SimpMessagingTemplate ws;

    @KafkaListener(topics = "${app.kafka.topics.odds:odds.updates}", groupId = "odds-ws")
    public void onOdds(OddsUpdated evt) {
        // fan-out odds to websocket topic per event/market
        String dest = "/topic/odds/" + evt.getEventId() + "/" + evt.getMarketId();
        ws.convertAndSend(dest, evt);
    }
}

File: sportsbook-app/odds-service/src/main/kotlin/com/sportsbook/oddsservice/websocket/OddsUpdateBroadcaster.kt
package com.sportsbook.oddsservice.websocket

import com.sportsbook.events.OddsUpdated
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.stereotype.Component

@Component
class OddsUpdateBroadcaster(
    private val ws: SimpMessagingTemplate,
) {

    @KafkaListener(topics = ["\${app.kafka.topics.odds:odds.updates}"], groupId = "odds-ws")
    fun onOdds(evt: OddsUpdated) {
        val dest = "/topic/odds/${evt.eventId}/${evt.marketId}"
        ws.convertAndSend(dest, evt)
    }
}
